### PR TITLE
Fix https://github.com/strongloop/loopback-next/issues/5259

### PR DESCRIPTION
**Pull request by raymondfeng**
Fixes https://github.com/strongloop/loopback-next/issues/5259

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: extensions/apiconnect/src/apiconnect.spec-enhancer.ts [hunk 1]
```diff
@@ -23,9 +29,18 @@ export class ApiConnectSpecEnhancer implements OASEnhancer {
 
   constructor(
     @config({optional: false}) private options: ApiConnectSpecOptions,
+    @inject(CoreBindings.APPLICATION_METADATA, {optional: true})
+    private appMetadata: ApplicationMetadata = {
+      name: 'LoopBack Application',
```
Comment (deepakrkris) [inline extensions/apiconnect/src/apiconnect.spec-enhancer.ts:34]: can we get the default value for x-ibm-name from the app's package.json

File: extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts [hunk 2]
```diff
@@ -42,6 +42,28 @@ describe('Extension for IBM API Connect - OASEnhancer', () => {
...
+    };
+    const spec = await server.getApiSpec();
+    expect(spec).to.containDeep(EXPECTED_SPEC);
+  });
+
+  it('adds default x-ibm-name to apiSpec info', async () => {
+    app.unbind(CoreBindings.APPLICATION_METADATA);
+    const EXPECTED_SPEC = {
+      info: {
+        'x-ibm-name': 'LoopBack Application',
```
Comment (deepakrkris) [inline extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts:60]: can we get the default value for x-ibm-name from the app's package.json

File: extensions/apiconnect/src/apiconnect.spec-enhancer.ts (hunk 1 above)
Comment (raymondfeng) [reply extensions/apiconnect/src/apiconnect.spec-enhancer.ts:34]: That's what the `inject` about. The `LoopBack Application` is the default value if application package is not loaded.

File: extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts (hunk 2 above)
Comment (raymondfeng) [reply extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts:60]: See above.

File: extensions/apiconnect/src/apiconnect.spec-enhancer.ts (hunk 1 above)
Comment (deepakrkris) [reply extensions/apiconnect/src/apiconnect.spec-enhancer.ts:34]: `CoreBindings.APPLICATION_METADATA` takes care of injecting the package json , is it ?

**Pull request merged**